numpy/_core/src/multiarray/ctors.c

```diff
@@ -4158,6 +4171,12 @@ _array_fill_strides(npy_intp *strides, npy_intp const *dims, int nd, size_t item
         for (i = nd - 1; i >= 0; i--) {
             strides[i] = itemsize;
             if (dims[i]) {
+                npy_intp tmp;
+                if (npy_mul_sizes_with_overflow(&tmp, (npy_intp)itemsize, dims[i])) {
+                    PyErr_SetString(PyExc_ValueError,
+                            "array is too big; overflow in _array_fill_strides.");
+                    return;
+                }
```

Comment:
Why? Can we please not do this?  I am happy to check ridiculous tuple lengths one could claim them to be *technically* valid also since it is local to the `__array_interface__`.
But this just can't be valid.  Someone might as well pass a bad data pointer if they run into this.

Reply:
@seberg 
Would it be ok to focus on the original bug fix in this PR and then I'll address the large tuple issue separately? 

I wasn't able to reproduce the large tuple issue, but I can investigate it further in another PR. 

I implemented your other suggestions in regards to converting `n` and `i` into type `Py_ssize_t`.